MapleServer2/PacketHandlers/Game/GuildHandler.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Collections.Generic;
+using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using Microsoft.Extensions.Logging;
+using MapleServer2.Types;
+
+namespace MapleServer2.PacketHandlers.Game
+{
+    public class GuildHandler : GamePacketHandler
+    {
+        public override RecvOp OpCode => RecvOp.GUILD;
+
+        public GuildHandler(ILogger<GuildHandler> logger) : base(logger) { }
+
+        private enum GuildMode : byte
+        {
+            Create = 0x1,
+            CheckIn = 0xF,
+            EnterHouse = 0x64,
+            GuildWindow = 0x54,
+            List = 0x55,
+            GuildNotice = 0x3E,
+            GuildDonate = 0x6E,
+        }
+
+        public override void Handle(GameSession session, PacketReader packet)
+        {
+            GuildMode mode = (GuildMode) packet.ReadByte();
+            switch (mode)
+            {
+                case GuildMode.Create:
+                    HandleCreate(session, packet);
+                    break;
+                case GuildMode.CheckIn:
+                    HandleCheckIn(session, packet);
+                    break;
+                case GuildMode.EnterHouse:
+                    HandleEnterHouse(session, packet);
+                    break;
+                case GuildMode.GuildWindow:
+                    HandleGuildWindowRequest(session, packet);
+                    break;
+                case GuildMode.List:
+                    HandleList(session, packet);
+                    break;
+                case GuildMode.GuildNotice:
+                    HandleGuildNotice(session, packet);
+                    break;
+                case GuildMode.GuildDonate:
+                    HandleGuildDonate(session, packet);
+                    break;
+                default:
+                    IPacketHandler<GameSession>.LogUnknownMode(mode);
+                    break;
+            }
+        }
+
+        private void HandleCreate(GameSession session, PacketReader packet)
+        {
+            string guildName = packet.ReadUnicodeString();
+
+            session.Player.Mesos -= 2000;
```

Comment:
A quick check if they have the mesos would be nice here, and you can just return if they don't have enough for now